### PR TITLE
If BORG_PASSPHRASE is present but wrong, don't prompt for password, f…

### LINKED ISSUE #791: How to check correctness of passphrase in a script?
I'm working on a shellscript that is run interactively which asks for the repo passphrase and then invokes borg multiple times. Is there a simple way to check in a script that the passphrase is correct?

My idea at first was to set `BORG_PASSPHRASE` and then run a simple command (`borg list`) and check the exit code, but unfortunately borg queries the tty for a passphrase if the one in the environment does not match. Right now I'm using _expect_ to get around this, but this is quite hackish -- is there a simpler method?

### PR DESCRIPTION
**Pull request by enkore**
…ail instead.

Leaves PassphraseKey alone, since I cannot test it.

Fixes #791

### CODE REVIEW

File: borg/key.py [hunk 1]
```diff
@@ -284,13 +288,19 @@ def detect(cls, repository, manifest_data):
...
+                for retry in range(0, 3):
+                    passphrase = Passphrase.getpass(prompt)
+                    if key.load(target, passphrase):
+                        break
+                else:
+                    raise PasswordRetriesExceeded
         else:
-            raise PasswordRetriesExceeded
+            if not key.load(target, passphrase):
+                raise PasswordWrong
```
Comment (ThomasWaldmann) [inline borg/key.py]: "Password" here vs. "Passphrase" elsewhere.

Comment (ThomasWaldmann) [inline borg/key.py]: hmm, nothing new though.

Comment (enkore) [inline borg/key.py]: I'll make it Passphrase here. Renaming existing output in a bugfix rel. is maybe not a good idea, so we'll do that in master later?

**Comment by Ape** [comment]
The commit subject line generally has a hard limit of 69 characters, but you should aim for 50.

Like for example:
- Fail if BORG_PASSPHRASE is present but wrong
- Do not prompt if BORG_PASSPHRASE is present but wrong

**Pull request merged**